### ISSUE TITLE
Our PyPI dev status is "Alpha"

### ISSUE THREAD

**Issue by lostanlen** [librosa/librosa#914]
#### Description

The Python Package Index (PyPI) lists seven stages for development status:
1. Planning
2. Pre-alpha
3. Alpha
4. Beta
5. Production/Stable
6. Mature
7. Inactive

Reference: https://pypi.org/classifiers/

The current status of librosa is "Alpha" (3).
In the https://pypi.org/project/librosa/

As of today the latest release of librosa is v0.7-rc1.
In the work towards a stable 0.7, perhaps we could bump this classifier to "Beta" (4), if not "Production/Stable" (5)?

**Comment by bmcfee** [comment]
> In the work towards a stable 0.7, perhaps we could bump this classifier to "Beta" (4), if not "Production/Stable" (5)?

I'm taking a much longer view of what "stable" means, and prefer to keep everything <1.0 in alpha state, not counting the immediate run up to a 1.0 release, which can be beta.  This is mainly motivated by wanting to stick to the spirit of [semantic versioning](https://semver.org/), allowing for API changes within minors at the 0.x stage.

What I'd like long term is to have a 1.0 release that's API-stable and properly obeys semantic versioning, and then we can start thinking about a 2.x series.  I don't know that we're far off from that point, but I haven't thought deeply about a checklist for 1.0, aside from the random issues we've collectively punted off to then via github tags.  It's possible (even likely) that 0.8 could become the beta for 1.0, since our rate of adopting new features has slowed considerably over the past couple of years.  We could then treat 1.0 as a feature-freeze after 0.8, and then punt radical changes to 2.x (eg going to a time-major instead of time-minor array shape, supporting multichannel directly, and so on).

**Comment by pawamoy** [comment]
Should we consider version 0.8 to be the beta of 1.0 then? I see the status is still Alpha for versions 0.8.x on PyPI.

**Comment by bmcfee** [comment]
> Should we consider version 0.8 to be the beta of 1.0 then? I see the status is still Alpha for versions 0.8.x on PyPI.

No, it should stay as alpha.  We've bumped up the priority on multi-channel support to the 0.9 release (rather than a 2.x series), and this will have some implications for the API.  Probably most things will stay backward-compatible if using mono signals, but I want to leave open the possibility of reworking some API things to be more channel-agnostic before locking down the API for 1.0.

Additionally, if the timing works out, we might want to push deprecation of the audioread backend up to 0.9 and remove it for the 1.0 release.  This is mainly a question of how quickly libsndfile gets updated to support mp3, but I would prefer to not have to support the audioread backend when librosa is stable.  This doesn't exactly change API, but it does play some strange tricks on our dependency chain and environment dependencies, and I'd like to have those things as stable as possible before marking a version as beta.

**Comment by pawamoy** [comment]
I see, thank you for the explanation and details 🙂
